### ISSUE TITLE
[req] MejorTorrent

### ISSUE THREAD

**Issue by ezruneko** [Jackett/Jackett#12559]
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
11-16 08:31:53 Info Starting Jackett v0.19.275 
11-16 08:31:54 Info Environment version: 5.0.12 (/app/Jackett/) 
11-16 08:31:54 Info OS version: Unix 5.4.140.1 (64bit OS) (64bit process) 
11-16 08:31:54 Info Jackett variant: CoreLinuxAmdx64 
11-16 08:31:54 Info File /etc/issue: Ubuntu 18.04.6 LTS \n \l 
11-16 08:31:54 Info Running in Docker: Yes (image build: v0.19.275-ls50)
```

### Description

- Add mejor torrent tracker
- Click on the test button

### Logged Error Messages

System.Exception: Test search in MejorTorrent => Found no results while trying to browse this tracker
   at Jackett.Common.Services.IndexerManagerService.TestIndexer(String name) in /home/vsts/work/1/s/src/Jackett.Common/Services/IndexerManagerService.cs:line 311
   at Jackett.Server.Controllers.IndexerApiController.Test() in /home/vsts/work/1/s/src/Jackett.Server/Controllers/IndexerApiController.cs:line 132
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jackett.Server.Middleware.CustomExceptionHandler.Invoke(HttpContext httpContext) in /home/vsts/work/1/s/src/Jackett.Server/Middleware/CustomExceptionHandler.cs:line 61

**Comment by garfield69** [comment]
This occurs quite regularly, and is caused by the website's recent-torrents page being empty.
![image](https://user-images.githubusercontent.com/6933240/141941219-12c7c360-9f1b-459c-ab66-ed8cbfc0675d.png)
There is nothing we can do about it except wait for it to populate again, which it will eventually do over the next weeks or so.
We use this page whenever you do a keywordless search, as using the search page without keywords is not allowed by the web site.

**Comment by garfield69** [comment]
Actually, I'm going to re-open this, and ask for a C# dev volunteer to add some code in the indexer, so that when this page returns no results, we perform a general search with the year instead.
This way we can still get a TEST to succeed and given this issue is a regular event it will be useful to have an work-around.

**Comment by mikeoscar2006** [comment]
Is it preferred to add the year-check in just the MejorTorrents indexer or maybe generally whenever the Test button is pressed on any indexer, it should try the current year in case of empty results?
Both are trivial so shouldn't be a problem so lmk @garfield69

**Comment by ilike2burnthing** [comment]
Just MejorTorrent, as it's the only one which fails regularly like this.

**Comment by ilike2burnthing** [comment]
Would it be possible to use the [main page](https://www.mejortorrentes.org) of the site instead of [Últimos Torrents](https://www.mejortorrentes.org/secciones.php?sec=ultimos_torrents)? Seems to serve a similar purpose.

I'm not sure this issue has raised its head again in the last 3 months, so this may not actually be necessary.

**Comment by mikeoscar2006** [comment]
You're right. I haven't faced the problem even once, so maybe we can re-open the issue if it pops up again.

**Comment by ilike2burnthing** [comment]
Yea, seems like a good idea. Could be they've resolved the issue on their end now. If not, we can reopen.

**Comment by garfield69** [comment]
![image](https://user-images.githubusercontent.com/6933240/154873950-21756cec-0b83-4e7c-9706-8ae7b68318eb.png)

**Comment by ilike2burnthing** [comment]
Ha! Spoke too soon. Main page it is then.

**Comment by killo3967** [comment]
I would appreciate it if you could reopen the case again since the problem persists and it is the only page in Spanish that is worthwhile. The idea of analyzing the main page is very good. Maybe the only option.

**Comment by ilike2burnthing** [comment]
It is open. Just waiting on @mikeoscar2006

**Comment by garfield69** [comment]
for clarity, the reason the https://www.mejortorrentes.org/secciones.php?sec=ultimos_torrents page is empty is that the web site has not had any new torrents since 17 feb 22 01:42:13 as shown on the home page https://www.mejortorrentes.org/
One wonders if the site is headed for oblivion.

**Comment by killo3967** [comment]
I don't know if what I'm about to say is relevant or not to the problem of mejortorrent. The thing is that I have this tracker configured in Jackett and when I do a test I get an error because there is nothing in https://www.mejortorrentes.org/secciones.php?sec=ultimos_torrents.
But the thing is that when I do a search it does respond correctly, both from Jackett and from Sonarr. In fact both Sonarr and Radarr have downloaded some "stuff" from this tracker. Don't you think it's strange ?

**Comment by garfield69** [comment]
> Don't you think it's strange ?

The issue is simply that the web site is not adding new torrents to its collection. The last torrent added was back on 17th February. 
So the page that is reports the latest updates usually presents up to 5 days worth of recent additions, but is empty.
I think the site is broken, and may soon be dead, relegated tothe role of an archive repository.
When you add the indexer for the first time, Jackett performs a TEST to check the site is set up.
TEST is just a keywordless search. The indexer is currently set up to fetch torrents from the sites latest-torrents page, and since this is empty the indexer fails the TEST.
For those using an already configured indexer, when you do a keyword search then you are delving into the search page, which returns any torrents the site already has.

Basically this indexer is very close to being removed altogether IMO since a site that does not provide new torrents is useless.

**Comment by garfield69** [comment]
Given that the site is no longer adding new torrents, and finding a C# dev to work on this is proving difficult, I will hack a temporary solution to allow the indexer to pass the TEST.
If a keywordless search is performed, I will just hardcode the indexer to do a search for `Historia`, which seems a good general word that encompasses all 3 categories, TV, Movies, and Audio.
If anyone can find a better word that returns few results for all 3 cats in one search, feel free to suggest it here.
The goal is to make the indexer respond as quickly as possible to a TEST.
In the event that the site miraculously returns to full operation, then we can revert this hack to restore normal functions of the indexer.

**Comment by garfield69** [comment]
https://www.mejortorrentes.org/ returning Error 520 Web server is returning an unknown error
the proxy alternates are returning 522 or 520 too.

**Comment by garfield69** [comment]
https://mejortorrent.nocensor.sbs/ redirects to https://mejortorrent.cc/ which appears to have updates for 12th may
unblockit still points to the broken site.
I'll try amending the indexer to handle this new domain

**Comment by garfield69** [comment]
*.cc is clearly a clone, using a different search path `?s=harry instead of `secciones.php?sec=buscador&valor=harry`
also the [Últimos Torrents](https://mejortorrent.cc/ultimos-torrents-3/) link returns the home page,
however trying `ultimos-torrents` returned https://mejortorrent.cc/ultimos-torrents-4-x/ which seems to work.

**Comment by garfield69** [comment]
Nah. the site is just too different and I am not up to completely rewriting the indexer to manage the multiple changes.
I'm going to remove this indexer and call it a day.

**Comment by garfield69** [comment]
v0.20.1066

**Comment by albertodiazsaez** [comment]
It's a shame this indexer had to be removed, as is one of the main ones for Spanish content, right now it's URL seems to be [mejortorrent.wtf](https://mejortorrent.wtf/), and it (without checking) should not work with the C# previously removed, and even though I'm not a main C# developer, I would like to try and maintain this tracker.

**Comment by garfield69** [comment]
I'll test the old indexer source against the domain you provided and if it works I'll restore the indexer to jackett.

**Comment by garfield69** [comment]
Nope, https://mejortorrent.wtf/ is also a clone and not the original under a new domain.
This site has different paths and layout to the original source and will require rewriting the indexer.
PR welcome.

**Comment by albertodiazsaez** [comment]
Maybe there has been a refactoring in the website and now it works like this, because I think this one and the .cc ones worked like that, I can try to rewrite the indexer, is there a guide or example indexer?

**Comment by garfield69** [comment]
the source can be restored by reverting https://github.com/Jackett/Jackett/commit/b2274030e7a16fd85ad884515122e2ada13f0376
and
https://github.com/Jackett/Jackett/commit/8e8ff4b5977cd7ddf10abc4eb4576e495ffefe63
after that its just a matter of recoding to suit the new domain

**Comment by IberianSoldierPC** [comment]
Still doesn't work atm

**Comment by IberianSoldierPC** [comment]
![Captura de pantalla 2023-01-31 181344](https://user-images.githubusercontent.com/56433990/215834025-7252b5b4-0eb1-452f-9fcf-d5ee6651138c.png)

**Comment by garfield69** [comment]
@IberianSoldierPC which *arr app are you using?

**Comment by IberianSoldierPC** [comment]
> @IberianSoldierPC which *arr app are you using?

Sonarr to be specific

**Comment by garfield69** [comment]
I got Sonarr to pass the indexer test by setting the Sonarr indexer config to TV(5000), TV/SD(5030), TV/HD(5040), Movies(5000)
But when searching for the first episode of South of Hell Sonarr finds no results.
This is because jackett generates https://mejortorrent.wtf/busqueda/page/1?q=South+of+Hell+S01E01
and the mejortorrent website only supports looking for titles without the season/episode tag.
![image](https://user-images.githubusercontent.com/6933240/215842483-f6b3ab9b-d31d-476d-b2cf-e90463696d8a.png)

So the site is not suitable for use with Sonarr ATM, and the indexer would need reworking to drop the SxxExx from the query to have any chance of fetching results for TV.

using the jackett search without SxxExx gets results
![image](https://user-images.githubusercontent.com/6933240/215842274-6de419ca-5df0-4f1a-8eef-8ea840231554.png)

**Comment by garfield69** [comment]
I think I can strip the SxxExx from the query so results are returned. I will work on this.

**Comment by garfield69** [comment]
I can now use Sonarr to find the first season first episode of South of Hell, so job done. Will be available in the next Jackett release due out in about 6 hours or so.

**Comment by garfield69** [comment]
v0.20.2814

**Comment by artai94** [comment]
> I can now use Sonarr to find the first season first episode of South of Hell, so job done. Will be available in the next Jackett release due out in about 6 hours or so.

Thanks to the new update you made, it works in the Jackett Manual search, but still no search results for me in Sonarr with this indexer. I'm at v0.20.2814

**Comment by IberianSoldierPC** [comment]
> v0.20.2814

Still can't add it to sonarr. I'd love to help if I could so thanks in advanced for your job!

**Comment by artai94** [comment]
> > v0.20.2814
> 
> Still can't add it to sonarr. I'd love to help if I could so thanks in advanced for your job!

Hey dude. I think i found the problem. Some TV show episode files are too small for the default HDTV-720p. look:
![image](https://user-images.githubusercontent.com/124089215/216089132-08d2341d-5380-43b4-a098-22b4f71c0a02.png)

If you make a manual search (click the person in the S01E01 at the right and click in the search tab), there you have the errors:
![image](https://user-images.githubusercontent.com/124089215/216091035-1a4912d6-e35a-4f8f-834e-20564305b960.png)


After I set up the minimum size to 0, the results started to appear: 
![image](https://user-images.githubusercontent.com/124089215/216089636-158d25c6-0e38-4d12-ac79-1907295cf5a2.png)

**Comment by IberianSoldierPC** [comment]
> **I got Sonarr to pass the indexer test by setting the Sonarr indexer config to TV(5000), TV/SD(5030), TV/HD(5040), Movies(5000)** 

Doing this fixed the problem. A true hero. The problem comes when trying to add the rest of the categories "Serie" and "Serie HD"

**Comment by artai94** [comment]
> > **I got Sonarr to pass the indexer test by setting the Sonarr indexer config to TV(5000), TV/SD(5030), TV/HD(5040), Movies(5000)**
> 
> Doing this fixed the problem. A true hero. The problem comes when trying to add the rest of the categories "Serie" and "Serie HD"
De nada, tio
Anyways, I've checked almost all of the boxes for the search and it seems to search only in the TV(5000), TV/SD(5030), TV/HD(5040), so... meh
![image](https://user-images.githubusercontent.com/124089215/216107570-9bc85c2f-ee8a-4ab3-98b6-4f3a7eb8ce19.png)

**Comment by garfield69** [comment]
the mejortorrent web site is barely suitable for indexer with jackett. The original author of the indexer has had to make compromises to make it work.
Since mejortorrent does not provide any file sizes for the torrents, the indexer sets some default values based on the category tag, 1.5GB, 1GB, 512MB, 100MB which is better than nothing but there is no way to improve this.
As for the Sonarr categories, for mejortorrent indexer in particular, only add the main cats I listed earlier (numbers in the 1000-9999 range).
The other cats (those with a 100000+number) are custom cats, and are sub categories of the main cats, so you do not need them in mejortorrent to fetch the available torrents.
